02/app.js

```diff
@@ -1,32 +1,56 @@
-document.addEventListener('DOMContentLoaded', init);
+document.addEventListener('DOMContentLoaded', init)
 
 function init() {
-    const divList = document.querySelectorAll('div');
-    
-    setBorderColorAsync(divList[0], 'red', function() {
-        setBorderColorAsync(divList[1], 'blue', function() {
-            setBorderColorAsync(divList[2], 'green', function() {
-                console.log('finish');
-            });
-        });
-    });
+	const divList = document.querySelectorAll('div')
 
+	setBorderColorAsync(divList[0], 'red')
+		.then(() => setBorderColorAsync(divList[1], 'blue'))
+		.then(() => setBorderColorAsync(divList[2], 'green'))
+		.then(() => console.log('finish'))
+
+	// rozumiem że finally by tutaj nie pasowało?
+	// .finally(() => console.log('finish'))
```

Comment:
To trochę inny przypadek - nas interesowało poprawne zakończenie (`.then`) zamiast dowolne (`.finally`) - czyli również niepoprawne ;)